.github/workflows/tests.yml

```diff
@@ -126,6 +162,10 @@ jobs:
     steps:
       - name: Checkout Amplify UI
         uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event_name == 'push' && github.ref || github.event.pull_request.head.ref }}
```

Comment:
Can you add a comment on this logic?

It reads like `github.ref` is correct when we push, but PRs rely on the `head.ref` on what to test?

Reply:
Good point! Yes, that's it. `github.ref` on `pull_request_target` always points to the target branch (`main`), so we need to make it a bit more explicit.

Reply:
https://github.com/aws-amplify/amplify-ui/pull/855/commits/76552aa0891bcba68de38fea6cf63af158866b23